### PR TITLE
WIP: Move eviction job to a swap environment.

### PR DESCRIPTION
**Pull request by mmiranda96**
Use a swap environment for eviction tests.

In order to run eviction tests with swap enabled, we need a swap environment. These are the main reasons to use the same job for existing eviction tests and future eviction-on-swap tests:
- They cover the same logic.
- Running non-swap test under such environment is ok, since access to swap is limited by a featureGate.
- We can reduce the number of jobs to be maintained and monitored.

Currently swap is only configured on Ubuntu, I still need to figure out if configuring it on COS is the same or if it requires some additional setup.

This work is related to https://github.com/kubernetes/kubernetes/issues/105023

/cc @ehashman @endocrimes
/kind feature
/priority important-soon
/sig node

### CODE REVIEW

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml [hunk 1]
```diff
@@ -264,7 +264,7 @@ periodics:
       - --deployment=node
       - --gcp-project-type=node-e2e-project
       - --gcp-zone=us-west1-b
-      - --node-args=--image-config-file=/workspace/test-infra/jobs/e2e_node/image-config-serial.yaml
+      - --node-args=--image-config-file=/workspace/test-infra/jobs/e2e_node/image-config-serial-swap.yaml
```
Comment (ehashman) [inline config/jobs/kubernetes/sig-node/node-kubelet.yaml:267]: Any reason to not reuse https://github.com/kubernetes/test-infra/blob/cc714da33d7ba85672aa4c7f58e0b3993155176d/jobs/e2e_node/swap/image-config-swap.yaml ?

**Review by ehashman — COMMENTED** [review]
> Currently swap is only configured on Ubuntu, I still need to figure out if configuring it on COS is the same or if it requires some additional setup.

We also have it available on Fedora nodes.

Please rebase!

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml (hunk 1 above)
Comment (mmiranda96) [reply config/jobs/kubernetes/sig-node/node-kubelet.yaml:357]: It looks like I replaced the config on all jobs using it, so there's no point on having two separate ones. Merging them...

**Comment by ehashman** [comment]
Closing this as a nudge to rebase... please reopen when ready for review!

/close

**Pull request closed without merging**